### ISSUE TITLE
OData generating subqueries

### ISSUE THREAD

**Issue by matheusdarosa** [OData/WebApi#715]
I am executing a simple listing operation where I return an IQueryable from Entity Framework context but when OData query the IQueryable it is generating a sub-query on the database. Please check the SQL code generated by EF:

```
SELECT Project1.created, 
       Project1.id, 
       Project1.utctime, 
       Project1.name, 
FROM(SELECT Extent1.id,
               Extent1.utctime,
               Extent1.created,
               Extent1.name,
...
FROM   person AS Extent1
WHERE  ( Extent1.id = @p__linq__0 )
OR( ( Extent1.id = @p__linq__1 ))) AS Project1 
ORDER BY Project1.created ASC
LIMIT  21
```

If I enumerate the IQueryable before the "ApplyTo" of OData then Entity Framework executes a simple query.

The problem here is that the "internal" is taking too long(200,000 rows) since the LIMIT is applied only on the outside query.

Any idea what could cause this behavior?

**Comment by VikingsFan** [comment]
@matheusdarosa if you enumerate the IQueryable before the "ApplyTo" of OData, OData will just apply the queryoption to the in-memory result from EF, so the query is original simple query, or OData will add queryoption to the original query, I didn't understand the `internal is taking too long` means, can you share more detail of your request url? I see there is a Orderby in the query, so it has to apply to all 200,000 rows then we can limit the result.

**Comment by matheusdarosa** [comment]
@VikingsFan I enumerated the IQueryable just to make a test. In the real case it is passed directly to the ApplyTo. What I tried to mean is the "Extent1" query is not applying the LIMIT, so it is taking a lot of time to load the data and when it finally loads than the "Project1" query returns 21 rows. 

But after some search I believe the problem is not with OData but with the MySql Connector. I found this issue:

https://bugs.mysql.com/bug.php?id=75272

As soon I have more information I will update this issue.
